pkg/workloadmeta/types.go

```diff
@@ -75,6 +75,10 @@ type Store interface {
 
 	// Dump lists the content of the store, for debugging purposes.
 	Dump(verbose bool) WorkloadDumpResponse
+
+	// WaitForCollectors waits until all collectors either start collecting
+	// data, are removed due to being disabled, or ctx is cancelled.
```

Comment:
I would think you would want to wait until the collectors have completed their first collection - not just until they've started?

Reply:
True. That's effectively the same for the kubelet collector (it pulls on `Start`), so I'll amend the doc to document that expectation (and fix non-compliant collectors).